src/auth/scheme/validate_external.ts

```diff
@@ -0,0 +1,44 @@
+import * as Hapi from 'hapi';
+import * as Boom from 'boom';
+import { compare } from 'bcrypt';
+
+type ExternalStrategyResponse =
+  Promise <{ isValid: boolean, credentials: { scope: string } | {} } | Boom<null>>;
+
+type ValidateExternal = (request: Hapi.Request, token: string, h: Hapi.ResponseToolkit)
+  => ExternalStrategyResponse;
+
+type TokenResponse = { api_token: string, api_token_access: string };
+
+
+const findMatching = async (xs: TokenResponse [], token: string): ExternalStrategyResponse => {
+  if (xs.length < 1) return { isValid: false, credentials: {} };
+
+  const { api_token: apiToken = null, api_token_access: scope = null } = xs[0];
+
+  if (!apiToken || !scope) throw new Error('Error collecting api token data from database');
```

Comment:
Both these fields are marked `NOT NULL` in the database so it doesn't seem necessary to make this check here.